crypto/dh/dh_pmeth.c

```diff
@@ -463,11 +463,10 @@ static int pkey_dh_derive(EVP_PKEY_CTX *ctx, unsigned char *key,
         if (*keylen != dctx->kdf_outlen)
             return 0;
         ret = 0;
-        Zlen = DH_size(dh);
-        Z = OPENSSL_malloc(Zlen);
-        if (Z == NULL) {
-            goto err;
-        }
+        if ((Zlen = DH_size(dh)) <= 0)
+            return 0;
+        if ((Z = OPENSSL_malloc(Zlen)) == NULL)
+            return 0;
```

Comment:
Suggestion: may be add a call to `ERR_raise(ERR_LIB_BN, ERR_R_MALLOC_FAILURE);` here.

Reply:
DWIW: I've added the error message.  There are other cases in the code where errors aren't fully reported and where errors aren't possible but are checked for.